rp2040-hal/src/timer.rs

```diff
@@ -289,6 +311,45 @@ macro_rules! impl_alarm {
                 }
             }
 
+            /// Schedule the alarm to be finished at the given timestamp. If [enable_interrupt] is
+            /// called, this will trigger interrupt `
+            #[doc = $int_name]
+            /// ` whenever this timestamp is reached.
+            ///
+            /// The RP2040 has been observed to take a little while to schedule an alarm. For this
+            /// reason, this function requires the timestamp to be at least 10 microseconds in the
+            /// future.
+            ///
+            /// The RP2040 is unable to schedule an event taking place in more than
+            /// `u32::max_value()` microseconds.
+            ///
+            /// [enable_interrupt]: #method.enable_interrupt
+            fn schedule_at<const FREQ_HZ: u32>(
+                &mut self,
+                timestamp: TimerInstantU64<FREQ_HZ>,
+            ) -> Result<(), ScheduleAlarmError> {
+                const MIN_MICROSECONDS: u64 = 10;
+                let timestamp = timestamp.ticks();
```

Comment:
Perhaps I don't understand the `fugit` API, but doesn't that only work if `FREQ_HZ == 1_000_000`?
If so, `timestamp` should be a `TimerInstantU64<1_000_000>`.

Reply:
It should Provided that the watchdog was configured properly.
It'd also make sense to provide an alias to this fully defined type.
I'll update accordingly.

Reply:
I think we can assume that the watchdog is properly configured at this point. Otherwise the whole API doesn't make much sense. get_counter also assumes a 1MHz tick.